### ISSUE TITLE
Powers incorrect for floating-point and rational power

### ISSUE THREAD

**Issue by dpsanders** [dpsanders/ValidatedNumerics.jl#109]
Consider the following
```julia
julia> x = @biginterval(27)
[2.700000000000000000000000000000000000000000000000000000000000000000000000000000e+01, 2.700000000000000000000000000000000000000000000000000000000000000000000000000000e+01]₂₅₆

julia> x^(1/3)
[2.999999999999999817044301297678855839521013101761257341876944169485799652263943, 3.000000000000000365911397404642321793745661450995000443687064161028801851381282]₂₅₆

julia> x^(1//3)
[2.999999999999999817044301297678855839521013101761257341876944169485799652263943, 3.000000000000000365911397404642321793745661450995000443687064161028801851381282]₂₅₆

julia> exp((1//3)*log(x))
[2.999999999999999999999999999999999999999999999999999999999999999999999999999931, 3.000000000000000000000000000000000000000000000000000000000000000000000000000069]₂₅₆
```

This shows that the current implementation of power with rational and real exponents is too pessimistic. For real powers `x`, the problem apparently lies in taking `xx = @interval(x)` [here](https://github.com/dpsanders/ValidatedNumerics.jl/blob/42bd3379e8bd6fbf7d543bd35aad164e0ce371bf/src/intervals/functions.jl#L96), and for rational powers `x` converting it to a real [here](https://github.com/dpsanders/ValidatedNumerics.jl/blob/42bd3379e8bd6fbf7d543bd35aad164e0ce371bf/src/intervals/functions.jl#L132).

The first problem can be removed by replacing it with `xx = convert(Interval{BigFloat}, x)`; this seems to also solve the second (although it shouldn't completely?)

In fact, according to [this article](http://link.springer.com/article/10.1007%2Fs00500-013-1008-8) by @oheim, as far as I understand it, just using `exp(x*log(a))` is guaranteed to be *almost* as good for `Interval{Float64}`, so probably we should just go with that. cc @lbenet

**Comment by oheim** [comment]
For ```x^(1//n)``` you could compute the _n_ th root. Apart from that, computing a correctly rounded power is difficult (see that article or [my thesis](http://exp.ln0.de/heimlich-power-2011.htm)).

Of course you could always use the exp(y*log(x)) approach, but it should be better to use the correctly rounded power function from MPFR (which you do with ```xx = convert(Interval{BigFloat}, x)```?).

**Comment by dpsanders** [comment]
Thanks @oheim. Yes, the idea of `xx = convert(Interval{BigFloat}, x)` is to take the smallest interval of MPFR floats (with the current global precision) that contains the real or rational `x`, and then compute `a^xx`, so an interval power of an interval, that is thus guaranteed to contain the correct result.

Then we use the correctly-rounded MPFR power function to compute the interval power of an interval. In my tests that is slightly (but only slightly) better than using `exp(xx*log(a))`.

I had thought about separately calculating the $n$th root using `mpfr_root`,but perhaps it's more work than it's worth.

**Comment by lbenet** [comment]
Thanks for sharing your paper and thesis. Reading the paper, I one should try to implement **pow3** for intervals. Am I right? (I haven't finished the paper yet...)

**Comment by dpsanders** [comment]
Fixed by #110
